app/controllers/shelters_controller.rb

```diff
@@ -14,8 +14,16 @@ def create
               state: params[:state],
               zip: params[:zip]
               })
-    shelter.save
-    redirect_to '/shelters'
+              
+    missing_fields = shelter_params.select{|_,v| v.nil? || v == ""}.keys
+    if missing_fields.empty?
+      shelter.save
+      redirect_to '/shelters'
+    else
+      flash[:notice] = "Unsuccessful shelter submission, please fill in the following fields prior to submission: #{missing_fields.each { |field| p "#{field} "}}"
```

Comment:
I think in terms of SRP and just clean code, the iteration over missing fields should probably be separated out of the flash notice itself, and then the flash notice can interpolate a variable that holds the value of the list of missing fields. 

I also wonder how we can get the list of fields to appear not in brackets like it seems they will based on the specs.

Maybe we can come back to this after finishing the rest of the user stories